jaxrs/WFLY-12779_Use-MicroProfile-Config-in-RESTEasy

```diff
@@ -0,0 +1,77 @@
+= Use MicroProfile Config in RESTEasy
+:author:            Ron Sigal
+:email:             rsigal@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, RESTEasy 3.x (targeted to Wildfly) uses servlet context-params to obtain
+configuration information. In RESTEasy 4.x, we have updated configuration to access
+all standard MicroProfile Config sources. Also, we have added three additional ConfigSources:
+
+* servlet init-params [ordinal 60]
+* filter init-params [ordinal 50]
+* servlet context-params [ordinal 40]
+
+In RESTEasy 4.x, all parameters, both RESTEasy parameters and application parameters, are
+available from all ConfigSources to both RESTEasy and application code. In RESTEasy 3.x, we
+propose to add a RESTEasy parameter, "resteasy.config.resteasy.parameters.only", which, when
+set to "true", will limit the three RESTEasy ConfigSources to returning RESTEasy parameters
```

Comment:
What parameters would be actually excluded? Can you outline an use-case?

Reply:
@ronsigal I find that confusing as well.  Do I understand it correctly, when `resteasy.config.resteasy.parameters.only=true` 

```
* servlet init-params [ordinal 60]
* filter init-params [ordinal 50]
* servlet context-params [ordinal 40]
```

parameters will be exposed only to RestEasy library? Why not to have all parameters available everywhere? Is it to keep backward compatibility of servlet context-params behaviour? Was there before this RFE way for application to consume servlet context-param?

Reply:
Removed reference to "resteasy.config.resteasy.parameters.only"